File: src/test/java/com/raychenon/leetcode/PossibleBipartitionTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 29/5/2020
 * https://leetcode.com/problems/possible-bipartition/
 */
public class PossibleBipartitionTest {

    private static PossibleBipartition instance;

    @BeforeClass
    public static void setUp() {
        instance = new PossibleBipartition();
    }

    @Test
    public void testExample1() {
        // 1 -> 3
        // 1 -> 2
        // 4  /
        int dislikes[][] = {{1, 2}, {1, 3}, {2, 4}};
        assertEquals(true, instance.possibleBipartition(4, dislikes));
    }

    @Test
    public void testExample2() {
        // 1 - 2
        //  |  |
        //   \ 3
        int dislikes[][] = {{1, 2}, {1, 3}, {2, 3}};
        assertEquals(false, instance.possibleBipartition(3, dislikes));
    }

    @Test
    public void testExample3() {
        // 1 - 2
        // 3 - 2
        // 3 - 4
        // 5 - 4
        // 1 - 5 , wrong 5 is already in the same group(or color) of 1
        int dislikes[][] = {{1, 2}, {2, 3}, {3, 4}, {4, 5}, {1, 5}};
        assertEquals(false, instance.possibleBipartition(5, dislikes));
    }


}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/PossibleBipartitionTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.TestCase
import org.junit.Test

/**
 * User: raychenon
 * Date: 29/5/2020
 * https://leetcode.com/problems/possible-bipartition/
 */
class PossibleBipartitionTest {

    @Test
    fun testExample1() {
        // 1 -> 3
        // 1 -> 2
        // 4  /
        val dislikes = arrayOf(intArrayOf(1, 2), intArrayOf(1, 3), intArrayOf(2, 4))
        TestCase.assertEquals(
            true,
            PossibleBipartition.possibleBipartition(4, dislikes)
        )
    }

    @Test
    fun testExample2() {
        // 1 - 2
        //  |  |
        //   \ 3
        val dislikes = arrayOf(intArrayOf(1, 2), intArrayOf(1, 3), intArrayOf(2, 3))
        TestCase.assertEquals(
            false,
            PossibleBipartition.possibleBipartition(3, dislikes)
        )
    }

    @Test
    fun testExample3() {
        // 1 - 2
        // 3 - 2
        // 3 - 4
        // 5 - 4
        // 1 - 5 , wrong 5 is already in the same group(or color) of 1
        val dislikes =
            arrayOf(intArrayOf(1, 2), intArrayOf(2, 3), intArrayOf(3, 4), intArrayOf(4, 5), intArrayOf(1, 5))
        TestCase.assertEquals(
            false,
            PossibleBipartition.possibleBipartition(5, dislikes)
        )
    }

}